mutalyzer/describe.py

```diff
@@ -1,837 +1,233 @@
+                    weight_position=extracted.weight_position))
+            elif variant.type & extractor.REVERSE_COMPLEMENT:
+                seq_list.append(ISeq(start=variant.transposition_start + 1,
+                    end=variant.transposition_end, reverse=True,
+                    weight_position=extracted.weight_position))
+            else:
+                seq_list.append(ISeq(
+                    sequence=s2[variant.sample_start:variant.sample_end],
+                    weight_position=extracted.weight_position))
+        elif not (variant.type & extractor.IDENTITY):
+            description.append(var_to_rawvar(s1, s2, variant,
+                weight_position=extracted.weight_position))
 
-    if not lcs_f_len : # Optimisation, not really needed.
-        return default
+        if variant.type & extractor.TRANSPOSITION_CLOSE:
+            in_transposition -= 1
 
-    r1_len = s1_end_f - lcs_f_len
-    r2_len = s1_end - s1_start - s1_end_f
-    m1_len = s2_end_f - lcs_f_len
-    m2_len = s2_end - s2_start - s2_end_f
+            if not in_transposition:
+                description.append(var_to_rawvar(s1, s2, variant, seq_list,
+                    weight_position=extracted.weight_position))
 
-    partial = protein_description(M, s1, s2, s1_start, s1_start + r1_len,
-        s2_start, s2_start + m1_len) + protein_description(M, s1, s2,
-        s1_end - r2_len, s1_end, s2_end - m2_len, s2_end)
+    if not description:
+        return Allele([DNAVar()])
+    return description
 
-    if alleleDescriptionLength(partial) <= alleleDescriptionLength(default):
-        return partial
-    return default
-#protein_description
 
-def describe(original, mutated, DNA=True):
+def describe_protein(s1, s2):
     """
-    Convenience function for DNA_description().
+    Give an allele description of the change from {s1} to {s2}.
 
-    @arg original:
-    @type original: unicode
-    @arg mutated:
-    @type mutated: unicode
+    :arg unicode s1: Sequence 1.
+    :arg unicode s2: Sequence 2.
 
-    @returns: A list of DescribeRawVar objects, representing the allele.
-    @rval: list(DescribeRawVar)
+    :returns list(RawVar): A list of RawVar objects, representing the allele.
     """
-    s1 = original
-    s2 = mutated
-    lcp = len(longest_common_prefix(s1, s2))
-    lcs = len(longest_common_suffix(s1[lcp:], s2[lcp:]))
-    s1_end = len(s1) - lcs
-    s2_end = len(s2) - lcs
-
-    if (s1_end - lcp) * (s2_end - lcp) > MAX_MATRIX_SIZE:
-        return
-
-    if not DNA:
-        M = LCS(s1, s2, lcp, s1_end, s2_end)
-        return protein_description(M, s1, s2, lcp, s1_end, lcp, s2_end)
-    #if
-    M = LCS(s1, s2, lcp, s1_end, s2_end, DNA=True)
-    return DNA_description(M, s1, s2, lcp, s1_end, lcp, s2_end)
-#describe
+    description = Allele()
+
+    fs1, fs2 = make_fs_tables(1)
+    longest_fs_f = max(find_fs(s1, s2, fs1), find_fs(s1, s2, fs2))
+    longest_fs_r = max(find_fs(s2, s1, fs1), find_fs(s2, s1, fs2))
```

Comment:
This code is now implemented in the description extractor, but we have not made a connection with Mutalyzer yet.

Of course I could clean this up, but this is currently unreachable code.